### PR TITLE
subclass agent to add setPersistSessionHandler

### PR DESCRIPTION
**Pull request by haileyok**
## Why

The API upgrade for OAuth deprecates `BskyAgent` as well as removes the ability to modify the persist session handler by calling `setPersistSessionHandler()` on the agent. Instead, the handler has moved to `SessionManager` and it can no longer be set.

To keep our previous logic in tact with minimal changes (and actually persist the session on change now 🙃) let's subclass `BskyAgent` to support this.

Also adding a `dispose()` method which unsets the session on the agent as well as unsets the persist handler (whereas before we made these two calls separately in `agent.ts`)

## Test Plan

Try changing your email on an alt, and it should immediately reflect the update.

### CODE REVIEW

File: src/state/session/agent-wrapper.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import {AtpSessionEvent, BskyAgent} from '@atproto/api'
+
+/**
+ * This is a temporary class that wraps `BskyAgent` and allows us to update the session persist handler when needed,
+ * since the deprecated `BskyAgent` after the API upgrades for OAuth no longer support setting the handler in the
+ * `SessionManager` itself.
+ */
+export class BskyAgentWrapper extends BskyAgent {
```
Comment (gaearon) [inline src/state/session/agent-wrapper.ts:8]: `BskyAppAgent`? To make it more legit.

File: src/state/session/agent-wrapper.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import {AtpSessionEvent, BskyAgent} from '@atproto/api'
+
+/**
+ * This is a temporary class that wraps `BskyAgent` and allows us to update the session persist handler when needed,
```
Comment (gaearon) [inline src/state/session/agent-wrapper.ts:4]: I wouldn't necessarily keep it temporary, it's nice to be able to preconfigure some things here. I might set up some other overrides for the online management.

**Review by gaearon — APPROVED** [review]

File: src/state/session/agent-wrapper.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      service,
+      persistSession: (event: AtpSessionEvent) => {
+        if (this.persistSessionHandler) {
+          this.persistSessionHandler(event)
+        }
+      },
+    })
+  }
+
+  setPersistSessionHandler(handler?: (event: AtpSessionEvent) => void) {
```
Comment (gaearon) [inline src/state/session/agent-wrapper.ts:23]: should probably be non-nullable here

Comment (haileyok) [reply src/state/session/agent-wrapper.ts:23]: Ah, yea might as well just use `dispose()` for that when needed 👍

**Pull request merged**